lib/seahorn/BvOpSem2.cc

```diff
@@ -617,20 +634,42 @@ class OpSemVisitor : public InstVisitor<OpSemVisitor>, OpSemVisitorBase {
     // 2) call void asm sideeffect "",
     // "r,~{memory},~{dirflag},~{fpsr},~{flags}"(i8* %7) #5, !dbg !102, !srcloc
     // !103
+    // 3) %_72 = call i16 asm "rorw $$8, ${0:w}",
+    // "=r,0,~{cc},~{dirflag},~{fpsr},~{flags}"(i16 %_69) #11, !dbg !731,
+    // !srcloc !733
 
     InlineAsm *IA = cast<InlineAsm>(CS.getCalledValue());
     const std::string &AsmStr = IA->getAsmString();
     // only handle "" instruction template and one argument
-    if (!AsmStr.empty() || CS.getNumArgOperands() > 1) {
+    // or "rorw $$8, ${0:w}" and one argument
+    if ((!AsmStr.empty() && AsmStr.compare(0, 16, "rorw $$8, ${0:w}") != 0) ||
+        CS.getNumArgOperands() > 1) {
       LOG("opsem",
           ERR << "Cannot handle inline assembly: " << *CS.getInstruction());
       return;
     }
 
     if (IA->getConstraintString().compare(0, 5, "=r,0,") == 0) {
-      // copy input to output
-      Expr readVal = lookup(*CS.getArgument(0));
-      setValue(*CS.getInstruction(), readVal);
+      if (AsmStr.compare(0, 16, "rorw $$8, ${0:w}") == 0) {
+        // Check Flag resgisters
+        llvm::SmallVector<llvm::StringRef, 4> AsmPieces;
+        llvm::SplitString(StringRef(IA->getConstraintString()).substr(5),
+                          AsmPieces, ",");
+        // Try to replace a call instruction with a call to a bswap intrinsic
+        if (!clobbersFlagRegisters(AsmPieces) ||
+            !expandCallInst(*cast<CallInst>(CS.getInstruction()),
+                            [](CallInst &CI) {
+                              return IntrinsicLowering::LowerToByteSwap(&CI);
+                            })) {
+          LOG("opsem", ERR << "Cannot handle inline assembly of integer swap: "
+                           << *CS.getInstruction());
+          return;
+        }
+      } else {
+        // Copy input to output
+        Expr readVal = lookup(*CS.getArgument(0));
```

Comment:
For my own understanding,  if flag registers are clobbered, we don't
1. replace instruction with `bswap`... why? i.e. can we always use `byteswap` when we see `rorw`?
2. print a warning in the `else` branch. Should we let the user know something like "Cannot handle inline assembly: "?

Reply:
1. The x86 contains some specific intrinsics we need to handle. Otherwise, it won't work for several harnesses under a Linux machine.